src/components/ArticleLanguage.js

```diff
@@ -9,12 +9,12 @@ export const ArticleLanguage = ({ lang, title, close }) => {
   const i18n = useI18n()
   const [articleLang, setArticleLang] = useState(lang)
   const [items, query, setQuery, numOfLanglink] = useSearchArticleLanguage(articleLang, title)
-  const [, setNavigation, getCurrent] = useNavigation('ArticleLanguage', containerRef, 'y')
 
   if (!items.length && !query) {
-    return <Loading message={i18n.i18n('article-language-loading-message')} />
+    return <Loading message={i18n.i18n('article-language-loading-message')} onClose={close} />
   }
 
+  const [, setNavigation, getCurrent] = useNavigation('ArticleLanguage', containerRef, 'y')
```

Comment:
**_Programatically solve the problem in reducer pop:_** 
we can move this back above of Loading component when we have this merged https://github.com/wikimedia/wikipedia-kaios/pull/174

or;

**_Ideally, in theory, follows the right order,_**
we can leave it here, so we have the correct order of softkey Loading first then ArticleLanguage, as `useNavigation` has softkey with origin `ArticleLanguage`

Reply:
Is there an advantage to moving it back above `Loading` component?

Reply:
Code looks prettier? Maybe?

Reply:
Yea leaving it as it is right now with `Loading` first makes sense to me